### PR TITLE
schedule: fix panic when switching placement rules

### LINKED ISSUE #7414: panic in tests/pdctl/operator
### Which jobs are failing
```
[2023/11/22 07:04:19.297 +00:00] [FATAL] [log.go:87] [panic] [recover="\"invalid memory address or nil pointer dereference\""] [stack="github.com/tikv/pd/pkg/utils/logutil.LogPanic\n\t/home/runner/work/pd/pd/pkg/utils/logutil/log.go:87\nruntime.gopanic\n\t/opt/hostedtoolcache/go/1.21.4/x64/src/runtime/panic.go:920\nruntime.panicmem\n\t/opt/hostedtoolcache/go/1.21.4/x64/src/runtime/panic.go:261\nruntime.sigpanic\n\t/opt/hostedtoolcache/go/1.21.4/x64/src/runtime/signal_unix.go:861\ngithub.com/tikv/pd/pkg/schedule/checker.(*RuleChecker).CheckWithFit\n\t/home/runner/work/pd/pd/pkg/schedule/checker/rule_checker.go:142\ngithub.com/tikv/pd/pkg/schedule/checker.(*Controller).CheckRegion\n\t/home/runner/work/pd/pd/pkg/schedule/checker/checker_controller.go:103\ngithub.com/tikv/pd/pkg/schedule.(*Coordinator).tryAddOperators\n\t/home/runner/work/pd/pd/pkg/schedule/coordinator.go:301\ngithub.com/tikv/pd/pkg/schedule.(*Coordinator).checkSuspectRegions\n\t/home/runner/work/pd/pd/pkg/schedule/coordinator.go:212\ngithub.com/tikv/pd/pkg/schedule.(*Coordinator).PatrolRegions\n\t/home/runner/work/pd/pd/pkg/schedule/coordinator.go:170"]
```

### CI link
https://github.com/tikv/pd/actions/runs/6952490392/job/18916139666?pr=7403
https://github.com/tikv/pd/actions/runs/6953924973/job/18919849256?pr=7368

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Close #7414.

### What is changed and how does it work?

When we going to disable the placement rule, we might first check:

https://github.com/tikv/pd/blob/8e6dcfbad0392bbacfc6fa1611457f75a8f0f2b4/pkg/schedule/checker/checker_controller.go#L89

Before we change it, the value is true. But when we call:
https://github.com/tikv/pd/blob/8e6dcfbad0392bbacfc6fa1611457f75a8f0f2b4/pkg/schedule/checker/checker_controller.go#L102
https://github.com/tikv/pd/blob/8e6dcfbad0392bbacfc6fa1611457f75a8f0f2b4/pkg/schedule/checker/priority_inspector.go#L64-L74
The value can be changed to false. So the fit will be nil in this case.

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by lhy1024 — APPROVED** [review]

File: pkg/schedule/checker/rule_checker.go [hunk 1]
```diff
@@ -132,6 +132,11 @@ func (c *RuleChecker) CheckWithFit(region *core.RegionInfo, fit *placement.Regio
 		return
 	}
 
+	// the placement rule is disabled
```
Comment (lhy1024) [inline pkg/schedule/checker/rule_checker.go:135]: Do we need to pick it to other branches?

File: pkg/schedule/checker/rule_checker.go [hunk 2]
```diff
@@ -132,6 +132,11 @@ func (c *RuleChecker) CheckWithFit(region *core.RegionInfo, fit *placement.Regio
...
+	if fit == nil {
+		return
```
Comment (bufferflies) [inline pkg/schedule/checker/rule_checker.go:137]: Do we need to add one metrics for it or add log?

**Review by bufferflies — APPROVED** [review]

File: pkg/schedule/checker/rule_checker.go (hunk 2 above)
Comment (rleungx) [reply pkg/schedule/checker/rule_checker.go:137]: I think no need since it happens only once.

**Pull request merged**